docroot/modules/custom/mass_hierarchy/src/MassHierarchyBasedBreadcrumbBuilder.php

```diff
@@ -98,13 +123,22 @@ public function build(RouteMatchInterface $route_match) {
     // Pass in the breadcrumb object for caching.
     $ancestor_entities = $this->mapper->loadAndAccessCheckEntitysForTreeNodes($entity_type, $ancestors, $breadcrumb);
 
+    /** @var \Drupal\entity_hierarchy_microsite\Entity\MicrositeInterface; $nearest_microsite */
+    $nearest_microsite = $this->nearestMicrositeLookup->getNearestMicrosite($route_entity);
+    $microsite_root = $nearest_microsite ? $nearest_microsite->getHome() : NULL;
```

Comment:
Good opportunity for the [nullsafe operator](https://www.php.net/manual/en/language.oop5.basic.php#language.oop5.basic.nullsafe) here: 
```suggestion
    $microsite_root = $nearest_microsite?->getHome();
```